[html] Refactor tests

jugglinmike opened the pull request:
testharness.js was recently extended with an API to explicitly opt-in to
the "single page test" feature [1]. As per WPT RFC 28 [2], tests which
do not use this API and which do not declare any subtests will soon be
reported as a harness error.

Update some tests which previously opted in implicitly to use the new
API. Update others to instead declare a single subtest (so that they are
no longer single-page tests).

[1] https://github.com/web-platform-tests/wpt/pull/19449
[2] https://github.com/web-platform-tests/rfcs/blob/master/rfcs/single_test.md

jugglinmike committed:
[html] Refactor tests

testharness.js was recently extended with an API to explicitly opt-in to
the "single page test" feature [1]. As per WPT RFC 28 [2], tests which
do not use this API and which do not declare any subtests will soon be
reported as a harness error.

Update some tests which previously opted in implicitly to use the new
API. Update others to instead declare a single subtest (so that they are
no longer single-page tests).

[1] https://github.com/web-platform-tests/wpt/pull/19449
[2] https://github.com/web-platform-tests/rfcs/blob/master/rfcs/single_test.md

foolip commented:
Diffs look good:
https://wpt.fyi/results/html?diff&filter=ADC&run_id=344460013&run_id=347940002
https://wpt.fyi/results/?diff&filter=ADC&run_id=319250008&run_id=346240011
https://wpt.fyi/results/html?diff&filter=ADC&run_id=311760008&run_id=333030004

The only differences are MISSING &rarr; NOTRUN for two subtests, which looks correct:
https://wpt.fyi/results/html/semantics/document-metadata/the-meta-element/pragma-directives/attr-meta-http-equiv-refresh/dynamic-append.html?run_id=319250008&run_id=346240011
https://wpt.fyi/results/html/semantics/embedded-content/media-elements/ready-states/autoplay-with-slow-text-tracks.html?run_id=311760008&run_id=333030004

@Hexcles this is a good example of needed to do quite a bit of work to understand the overall effect of a PR.

foolip commented on html/interaction/focus/the-autofocus-attribute/not-on-first-task.html:
```diff
@@ -12,6 +12,7 @@
 
 <script>
 "use strict";
+setup({ single_test: true });
```
I'd prefer `test(() => { ... })` wrapping here and for the other synchronous tests in this PR. (This is the first one.)

jugglinmike replied:
I've made the change you requested. Many of these tests (and still more in the related pull requests) could be expressed with a single `async_test` by swapping out `done()` for `t.done()`. Do you think that's preferable, too?

foolip commented on html/semantics/embedded-content/the-iframe-element/srcdoc_change_hash.html:
```diff
@@ -6,6 +6,8 @@
 <script src="/resources/testdriver.js"></script>
 <script src="/resources/testdriver-vendor.js"></script>
 <script>
+  setup({ single_test: true });
```
I'd prefer https://github.com/web-platform-tests/wpt/pull/19926 to this change.

jugglinmike replied:
Okay. I've merged your pull request and removed all modifications from this changeset so it will merge cleanly.

foolip reviewed (approved):
@jugglinmike this looks good now, leaving merging to you in case you want to tweak the commit message

jugglinmike commented:
Thanks! I've rebased and reworded the commit and updated the title/description of this pull request accordingly. [The original version of this branch is available here.](https://github.com/web-platform-tests/wpt/compare/master...bocoup:single-test-opt-in-html-orig) The latest version just passed in CI, so I'll do the honors.
